source/Camera/Basic/C++Halcon/HalconConnectUsingGenICam/HalconConnectUsingGenICam.cpp

```diff
@@ -0,0 +1,117 @@
+#include <Zivid/Zivid.h>
+#include <halconcpp/HalconCpp.h>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <chrono>
+#include <iostream>
+
+
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string filename) {
+	model.WriteObjectModel3d(HalconCpp::HString{ "ply" },
+		HalconCpp::HString{ filename.c_str() },
+		HalconCpp::HString{ "invert_normals" },
+		HalconCpp::HString{ "false" });
+}
+
+
+void setColorsInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const HalconCpp::HImage &RGB) {
+
+	HalconCpp::HRegion domain = RGB.AccessChannel(1).GetDomain();
+	HalconCpp::HTuple rows, cols;
+	domain.GetRegionPoints(&rows, &cols);
+
+	HalconCpp::HTuple Rtup = RGB.AccessChannel(1).GetGrayval(rows, cols);
+	HalconCpp::HTuple Gtup = RGB.AccessChannel(2).GetGrayval(rows, cols);
+	HalconCpp::HTuple Btup = RGB.AccessChannel(3).GetGrayval(rows, cols);
+
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+
+}
+
+HalconCpp::HTuple getFirstAvailableZividDevice(const HalconCpp::HTuple &devices) {
+	auto device = HalconCpp::HTuple();
+	auto tmpSplit = HalconCpp::HTuple();
+	auto splitDevices = HalconCpp::HTuple();
+
+	for (int i = 0; devices.Length(); i++) {
```

Comment:
We tend to use size_t in for loops elsewhere but I am not sure what is the reason. @eskaur can you help?

Reply:
This function, getFirstAvailableZividDevice(const HalconCpp::HTuple &devices), is a bit weird because HString and HTuples are not normal C++ strings or tuples. Therefore I cannot use standard C++ methods. 

What I can do is convert the Hstring to a normal string, do the interation and detect Zivid, and then create a new HString that is called 'Device' and is the input for openFramegrabber(...). Maybe this is more readable for a customer?

Reply:
Rewrote the entire function to a more simple one